<?xml version="1.0"?>
<robot name="visual">


  <material name="red">
    <color rgba="0.8 0 0 0.7"/>
  </material>
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.154 0.140 0.010"/>
      </geometry>
    </visual>
    <origin rpy="0 0 0" xyz="0 0 0.005"/>
  </link>

  <link name="trunk">
    <visual>
      <geometry>
      <cylinder length="0.064" radius="0.055"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.032"/>
      <material name="white"/>
    </visual>
  </link>


  <joint name="base_joint" type="revolute">
    <axis xyz=" 0 0 1"/>
    <parent link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.010"/>
    <limit effort="1000.0" lower="-2.6" upper="2.6" velocity=".5"/>
    <child link="trunk"/>
  </joint>

  <link name="upper_arm">
    <visual>
      <geometry>
    <box size="0.032 0.042 0.1005"/>
      </geometry>
      <origin rpy="0 1.570796 0" xyz=".05025 0 0"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="shoulder_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="trunk"/>
    <origin rpy="0 0 0" xyz="0.010 0 0.064"/>
    <limit effort="1000.0" lower="-3.14" upper="0" velocity=".5"/>
    <child link="upper_arm"/>
  </joint>

  <link name="forearm_z">
    <visual>
      <geometry>
    <box size="0.025 0.028 0.097"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.0379"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="elbow_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="upper_arm"/>
    <origin rpy="0 0 0" xyz="0.1005 0 0"/>
    <limit effort="1000.0" lower="0" upper="3.14" velocity=".5"/>
    <child link="forearm_z"/>
  </joint>
  
  <link name="forearm_x">
    <visual>
      <geometry>
    <box size="0.083 0.028 0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0275 0 0"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="forearm_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="forearm_z"/>
    <origin rpy="0 0 0" xyz="0 0 0.0758"/>
    <limit effort="1000.0" lower="-1.95" upper="1.95" velocity=".5"/>
    <child link="forearm_x"/>
  </joint>

  <link name="hand">
    <visual>
      <geometry>
    <box size="0.063 0.041 0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0205 0 0"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="wrist_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="forearm_x"/>
    <origin rpy="0 0 0" xyz="0.0627 0 0"/>
    <limit effort="1000.0" lower="-1.75" upper="1.75" velocity=".5"/>
    <child link="hand"/>
  </joint>

  <link name="gripper_base">
    <visual>
      <geometry>
    <box size="0.080 0.055 0.006"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.040 0 -0.0015"/>
      <material name="red"/>
    </visual>
  </link>  

  <joint name="fingers_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="hand"/>
    <origin rpy="0 0 0" xyz="0.0315 0 0"/>
    <limit effort="1000.0" lower="-1.95" upper="1.95" velocity=".5"/>
    <child link="gripper_base"/>
  </joint>

  <link name="gripper_temp">
    <visual>
      <geometry>
    <box size="0 0 0"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 00"/>
      <material name="white"/>
    </visual>
  </link>  

  <joint name="gripper" type="revolute">
    <axis xyz="0 0 1"/>
    <parent link="gripper_base"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="1000.0" lower="0" upper="1.58" velocity=".5"/>
    <child link="gripper_temp"/>
  </joint>

  </robot>
